src/SynthesiserRelation.cpp

```diff
@@ -1158,8 +1161,24 @@ void SynthesiserEqrelRelation::generateTypeStruct(std::ostream& out) {
     out << "}\n";
 
     // lowerUpperRange methods, one for each of the 4 possible search patterns
+    size_t arity = 2;
     for (int i = 1; i < 4; i++) {
-        out << "range<iterator> lowerUpperRange_" << i;
+        SearchSignature s(arity);
+        // if the bit is set then set it in the search signature
+        for (size_t j = 0; j < arity; j++) {
+            if (i & (1 << j)) {
+                s.set(j, AttributeConstraint::Equal);
+            }
+        }
+
+        if (i & 1) {
+            s.set(0, AttributeConstraint::Equal);
+        }
+        if (i & 2) {
+            s.set(1, AttributeConstraint::Equal);
+        }
```

Comment:
Are these lines intended to still be here? It looks like they duplicate the for loop before them.

Reply:
Great catch. I'll fix this up now.